pagefaultgames/pokerogue
# [QoL] Pity Toggle Setting

## What are the changes?
Added a pity toggle setting that can be enabled or disabled, and will accordingly enable or disable the gacha's pity code via a new extra conditional for the appropriate sections. Gacha pity is on by default.

## Why am I doing these changes?
Some players do not like the pity system, as it makes the game feel more like a typical grind, and takes away from the satisfaction of getting something rare from the gacha. By adding an option to turn it off or back on, these players can enjoy the game without dealing with the pity, while the players that want it can still have it, and people can turn it on or off as they wish if they change their mind.

## What did change?
There is a new option in the options menu that sets a new option variable called gachaPity to true or false based on whether on or off is selected, respectively. An extra condition on each of the sections of code correlating to the pity system has been added, both for species and egg rarity, that requires the value of the gacha pity option to be on (true) for those sections of code to execute. There are also some small updates to the code comments to reflect this.

### Screenshots/Videos
![temp2](https://github.com/pagefaultgames/pokerogue/assets/172436842/0606b62c-8d11-4c6d-ad79-243962747d35)
![temp](https://github.com/pagefaultgames/pokerogue/assets/172436842/a5f2014b-cec9-4bd7-b82c-5328aba5bb18)


## How to test the changes?
Add console logging outputs to desired locations around the new code, set the instant hatching override to true, run the game locally, and use breakpoints/console commands to acquire vouchers. It should be possible to verify functionality cases by turning the new option on/off and checking the console outputs.

## Checklist
